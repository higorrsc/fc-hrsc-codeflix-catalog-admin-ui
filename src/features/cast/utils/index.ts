import { CastMemberType } from '../types';

export function getCastMemberTypeOptions() {
  return Object.values(CastMemberType)
    .filter((value) => typeof value === 'number')
    .map((value) => ({
      value,
      label:
        CastMemberType[value as CastMemberType].charAt(0) +
        CastMemberType[value as CastMemberType].slice(1).toLowerCase(),
    }));
}
