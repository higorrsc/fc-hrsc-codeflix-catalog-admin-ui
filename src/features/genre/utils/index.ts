import { Category } from '../../category/types';
import { Genre } from '../types';

export const mapGenreToForm = (genre: Genre) => {
  return {
    id: genre.id,
    name: genre.name,
    is_active: genre?.is_active,
    categories_id: genre.categories?.map((category: Category) => category.id),
  };
};
