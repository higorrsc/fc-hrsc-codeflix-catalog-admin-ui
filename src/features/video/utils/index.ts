import { Video, VideoPayload } from 'src/features/video/types';

export const mapVideoToForm = (video: Video): VideoPayload => {
  return {
    id: video.id,
    title: video.title,
    description: video.description,
    year_launched: video.year_launched,
    opened: video.opened,
    rating: video.rating,
    duration: video.duration,
    genres_id: video.genres?.map((genre) => genre.id),
    categories_id: video.categories?.map((category) => category.id),
    cast_members_id: video.cast_members?.map((cast_member) => cast_member.id),
  };
};
