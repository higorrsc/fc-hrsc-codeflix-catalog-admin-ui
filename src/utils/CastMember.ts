import { CastMemberType } from '../types/CastMember';

const castMemberTypeLabelMap: Record<CastMemberType, string> = {
  [CastMemberType.ACTOR]: 'Actor',
  [CastMemberType.DIRECTOR]: 'Director',
};

export function getCastMemberTypeLabel(type: CastMemberType): string {
  return castMemberTypeLabelMap[type] || 'Unknown';
}

export const castMemberTypeOptions = [
  { value: CastMemberType.ACTOR, label: 'Actor' },
  { value: CastMemberType.DIRECTOR, label: 'Director' },
];
