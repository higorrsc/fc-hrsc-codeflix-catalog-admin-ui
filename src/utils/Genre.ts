import { Category } from '../types/Category';
import { Genre } from '../types/Genre';

export const mapGenreToForm = (genre: Genre) => {
  return {
    id: genre.id,
    name: genre.name,
    is_active: genre?.is_active,
    categories_id: genre.categories?.map((category: Category) => category.id),
  };
};
