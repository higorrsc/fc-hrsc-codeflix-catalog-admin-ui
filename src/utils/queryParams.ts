import { ListOptions } from '../types/_shared'; // Import ListOptions if needed for typing map

/**
 * Maps specific frontend parameter names (keys often matching ListOptions)
 * to their corresponding backend query parameter names expected by the API.
 * We use Partial<Record<keyof ListOptions | string, string>> to allow mapping
 * ListOptions keys and potentially other string keys if needed later.
 */
const paramKeyMap: Partial<Record<keyof ListOptions | string, string>> = {
  perPage: 'per_page',
  // Add other mappings if needed
};

/**
 * Parses an object containing potential query parameters into a URL query string
 * suitable for API requests.
 *
 * - Accepts an object like `ListOptions` or any object with potential query params.
 * - Filters out parameters with `null` or `undefined` values.
 * - Explicitly ignores UI-specific keys like `rowsPerPage`.
 * - Maps keys according to `paramKeyMap` (e.g., 'perPage' becomes 'per_page').
 * - Converts remaining parameter values to strings.
 *
 * @template T - The type of the params object (e.g., ListOptions).
 * @param params - An object containing potential query parameters.
 * @returns A URL query string (without the leading '?'). Returns an empty string
 *          if no valid parameters are found after filtering and mapping.
 */
export function parseQueryParams<T extends Record<string, any>>(
  params: T
): string {
  const queryParams = new URLSearchParams();

  Object.entries(params).forEach(([key, value]) => {
    // 1. Skip null/undefined/empty string for search
    if (
      value === null ||
      value === undefined ||
      (key === 'search' && value === '') // Also skip empty search strings
    ) {
      return;
    }

    // 2. Skip known UI-specific keys
    if (key === 'rowsPerPage') {
      return;
    }

    // 3. Determine the correct key name for the API
    const mappedKey = paramKeyMap[key as keyof typeof paramKeyMap] || key;

    // 4. Convert the value to a string and append
    queryParams.append(mappedKey, String(value));
  });

  return queryParams.toString();
}
