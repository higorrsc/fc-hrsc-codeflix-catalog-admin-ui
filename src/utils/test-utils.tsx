import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import React, { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';

import { ThemeProvider } from '@mui/system';
import { configureStore } from '@reduxjs/toolkit';
import { SnackbarProvider } from 'notistack';
import { BrowserRouter } from 'react-router-dom';
import type { RootState } from '../app/store';
import { appTheme } from '../config/theme';
import { apiSlice } from '../features/api/apiSlice';
import { castMembersApiSlice } from '../features/cast/castMemberSlice';
import { categoriesApiSlice } from '../features/categories/categorySlice';
// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: ReturnType<typeof configureStore>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  extendedRenderOptions: ExtendedRenderOptions = {}
) {
  const {
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        [categoriesApiSlice.reducerPath]: apiSlice.reducer,
        // @ts-ignore
        [castMembersApiSlice.reducerPath]: apiSlice.reducer,
      },
      // Adding the api middleware enables caching, invalidation, polling,
      // and other useful features of `rtk-query`.
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(apiSlice.middleware),
    }),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }: PropsWithChildren) => (
    <React.StrictMode>
      <BrowserRouter>
        <Provider store={store}>
          <ThemeProvider theme={appTheme}>
            <SnackbarProvider maxSnack={3}>{children}</SnackbarProvider>
          </ThemeProvider>
        </Provider>
      </BrowserRouter>
    </React.StrictMode>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
