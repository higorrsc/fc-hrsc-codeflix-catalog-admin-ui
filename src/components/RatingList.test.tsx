import { render, screen, within } from '../utils/test-utils'; // Assuming this provides ThemeProvider
import { RatingList } from './RatingList';

const ratingValues = ['L', '10', '12', '14', '16', '18'];

describe('Test RatingList Component', () => {
  it('should render all rating options correctly by default and match snapshot', () => {
    const { asFragment } = render(<RatingList />);

    const radioButtons = screen.getAllByRole('radio');
    expect(radioButtons).toHaveLength(ratingValues.length);

    ratingValues.forEach((value) => {
      // Check radio button by its value
      const radio = screen.getByDisplayValue(value) as HTMLInputElement;
      expect(radio).toBeInTheDocument();
      expect(radio).toBeEnabled();

      // Check the FormControlLabel wrapping the radio
      const formControlLabel = radio.closest('label');
      expect(formControlLabel).toBeInTheDocument();
      expect(formControlLabel).toHaveClass(
        'MuiFormControlLabel-labelPlacementTop'
      );

      // Check that the Rating component (identified by its text) is within the label
      if (formControlLabel) {
        expect(within(formControlLabel).getByText(value)).toBeInTheDocument();
      }
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render all rating options as disabled when isDisabled is true', () => {
    render(<RatingList isDisabled={true} />);

    ratingValues.forEach((value) => {
      const radio = screen.getByDisplayValue(value) as HTMLInputElement;
      expect(radio).toBeInTheDocument();
      expect(radio).toBeDisabled();
    });
  });

  it('should apply correct sx styling for the first and last items', () => {
    render(<RatingList />);

    // First item
    const firstRadio = screen.getByDisplayValue(ratingValues[0]);
    const firstLabel = firstRadio.closest('label');
    expect(firstLabel).toHaveStyle('margin-left: 0px');

    // Last item
    const lastRadio = screen.getByDisplayValue(
      ratingValues[ratingValues.length - 1]
    );
    const lastLabel = lastRadio.closest('label');
    expect(lastLabel).toHaveStyle('margin-right: 0px');

    // Check an intermediate item to ensure it doesn't have these specific overrides
    // (MUI might apply its own margins, we are checking our overrides)
    if (ratingValues.length > 2) {
      const middleRadio = screen.getByDisplayValue(ratingValues[1]);
      const middleLabel = middleRadio.closest('label');

      // Get computed style to check if the specific override is NOT there.
      // Note: This can be brittle if MUI default styles change.
      // It's often sufficient to just test that the overrides ARE applied where expected.
      const middleLabelStyles = window.getComputedStyle(middleLabel!);
      expect(middleLabelStyles.marginLeft).not.toBe('0px');
      // marginRight might be 0px by default in some flex scenarios or if it's the last in a row.
      // So, this assertion is less reliable without knowing MUI's exact default.
      // For now, focusing on the presence of overrides is safer.
    }
  });

  it('should ensure all labels have Rating components', () => {
    render(<RatingList />);
    ratingValues.forEach((value) => {
      // This implicitly checks that a Rating component rendering this text exists
      expect(screen.getByText(value)).toBeInTheDocument();
    });
  });
});
