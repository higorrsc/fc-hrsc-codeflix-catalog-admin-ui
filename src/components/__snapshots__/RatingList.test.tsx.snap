// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RatingList Component should render all rating options correctly by default and match snapshot 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop css-1nhsx05-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value="L"
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-pz392n"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          L
        </p>
      </div>
    </span>
  </label>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop css-1hgunqz-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value="10"
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-17d16e3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          10
        </p>
      </div>
    </span>
  </label>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop css-1hgunqz-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value="12"
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-vyctuh"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          12
        </p>
      </div>
    </span>
  </label>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop css-1hgunqz-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value="14"
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-1m4fxp9"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          14
        </p>
      </div>
    </span>
  </label>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop css-1hgunqz-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value="16"
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-1aa8pkh"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          16
        </p>
      </div>
    </span>
  </label>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementTop css-or594t-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value="18"
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      <div
        class="MuiBox-root css-12xjk60"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          18
        </p>
      </div>
    </span>
  </label>
</DocumentFragment>
`;
